sphinx/screens-detail/create-tribe/create-tribe/src/main/java/chat/sphinx/create_tribe/ui/CreateTribeFragment.kt

```diff
@@ -241,9 +247,43 @@ internal class CreateTribeFragment: SideEffectFragment<
                     viewModel.imageLoaderDefaults
                 )
             }
-            CreateTribeViewState.CreatingTribe -> {
+            CreateTribeViewState.SavingTribe -> {
                 binding.progressBarCreateTribe.visible
             }
+            is CreateTribeViewState.LoadingExistingTribe -> {
+                binding.progressBarLoadTribeContainer.visible
+            }
+            is CreateTribeViewState.ExistingTribe -> {
```

Comment:
DTO should not be utilized here. We need to get in the practice of the `ViewState` containing **only** the data necessary to populate the view.

A great way of thinking of it is by asking "When I call `viewModel.currentViewState`, what text is being presented in my `textViewTribeName` view?"

Having the Dto in the `ViewState` does not depict what the view's state is which is especially necessary in dynamically loaded values (non-static data from a data source).


Utilizing an operator function allows for dynamic overloading of the constructor that should help:
```
data class MyViewState private constructor(
    val name: String,
    val photoUrl: String?, // or utilize the PhotoUrl wrapper for type safety
): ViewState<MyViewState>() {

    companion object {
        operator fun invoke(myDto: MyDto): MyViewState =
            MyViewState(
                name = myDto.user_name,
                photoUrl = myDto.user_photo,
            )
    }
}

Reply:
Changes applied on EditTribe and JoinTribe fragments

Reply:
I didn't use the wrappers since that could produce some values not to show on the form in case they were somehow introduced in the wrong way on other platform (that could allow it), which would be confusing for the user.